lib/puppet/file_serving/terminus_selector.rb

```diff
@@ -21,6 +26,8 @@ def select(request)
       else
         Puppet[:default_file_terminus]
       end
+   when "http","https"
+     :http
```

Comment:
I'm sure once I've gotten farther into this code this will become apparent, but I'm curious why both this block and the previous one where you check the `request.key` are necessary?